client/src/main/java/com/boku/auth/http/client/BokuAPIClient.java

```diff
@@ -257,7 +268,7 @@ private BokuAPIClientResponse executeAndReturnAPIResponse() throws IOException,
 
             // Verify signature on the response
             if (authHeader != null) {
-                verifyResponseSignature(httpResponse, apiClientResponse);
+                verifyResponseSignature(httpResponse, apiClientResponse, this.requireSignedResponse);
```

Comment:
We want the logic to be,
 - If requireSignedResponse is true, require an X-SignedResponse header
 - If there is a X-SignedResponse header, verify it regardless

The old logic of checking for a request Authorization header to decide how to behave is now obsolete.

What I would do:
 - Get rid of the `authHeader != null` check
 - Move `Header[] respAuthHeaders = httpResponse.getHeaders(AuthorizationHeader.RESPONSE_HEADER)` out to this method.
 - If `requireSignedResponse` and there is no response header, produce the error from here.
 - If there is a response header, pass it into `verifyResponseSignature` - by the time that method is called, we know we require a valid signature, so it's either going to pass or throw an error.

Additionally you need to fix the flag setting behavior but I'll comment on that directly.